src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -0,0 +1,520 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import itertools
+import logging
+import os
+import re
+import shutil
+import time
+import traceback
+from contextlib import contextmanager
+from textwrap import dedent
+
+from pex.pex_info import PexInfo
+from six import StringIO
+from six.moves import configparser
+
+from pants.backend.python.python_setup import PythonRepos, PythonSetup
+from pants.backend.python.subsystems.pytest import PyTest
+from pants.backend.python.targets.python_tests import PythonTests
+from pants.backend.python.tasks2.gather_sources import GatherSources
+from pants.backend.python.tasks2.python_execution_task_base import PythonExecutionTaskBase
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError, TestFailedTaskError
+from pants.base.hash_utils import Sharder
+from pants.base.workunit import WorkUnitLabel
+from pants.build_graph.target import Target
+from pants.task.testrunner_task_mixin import TestRunnerTaskMixin
+from pants.util.contextutil import (environment_as, temporary_dir, temporary_file,
+                                    temporary_file_path)
+from pants.util.dirutil import safe_mkdir, safe_open
+from pants.util.process_handler import SubprocessProcessHandler
+from pants.util.strutil import safe_shlex_split
+
+
+# Initialize logging, since tests do not run via pants_exe (where it is usually done).
+logging.basicConfig()
+logger = logging.getLogger(__name__)
```

Comment:
The commit message mentions 
```
Caches also now use stdlib logging instead of context.log since the latter
has references to the runtracker making it almost impossible to pickle.
```